kafka-client/src/main/java/dev/responsive/kafka/internal/db/TimePartitionableKey.java

```diff
@@ -0,0 +1,11 @@
+package dev.responsive.kafka.internal.db;
+
+import java.sql.Time;
+
+public interface TimePartitionableKey<T> extends Comparable<T> {
+  /**
+   * @return the timestamp that partitioners can use to partition
+   *         keys of this type into different segments.
+   */
+  long getPartitionTimestamp();
```

Comment:
This is probably a bad name for this method.

Reply:
two notes on the naming:
1. style-wise, we use "get"-less getters -- eg `timestamp()` rather than `getTimestamp`
2. The term "partition" is massively overloaded right now so it's probably best to be specific or avoid it altogether. In this case the "partitions" are really just the "segments" -- it made more sense in the original C* implementation, where there was a concept of a partition in the C* table itself. Mongo doesn't exactly have partitions but the name stuck for things like the "SegmentPartitioner"
In this case, nothing feels grammatically perfect, but I'd suggest calling this method just `timestamp()` and then naming the class itself `SegmentKey` or `SegmentableKey` (is "segmentable" a word? definitely an awkward one if so...)
I think the important thing is really the javadocs (which are spot on here)